ta4j-core/src/main/java/org/ta4j/core/indicators/averages/JMAIndicator.java

```diff
@@ -0,0 +1,150 @@
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package org.ta4j.core.indicators.averages;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.ta4j.core.Indicator;
+import org.ta4j.core.indicators.CachedIndicator;
+import org.ta4j.core.num.Num;
+import org.ta4j.core.num.NumFactory;
+
+/**
+ * Jurik Moving Average (JMA) Indicator.
+ *
+ * JMA, or Jurik Moving Average, is a type of moving average developed by Mark
+ * Jurik. It is known for its ability to respond to price changes more smoothly
+ * than traditional moving averages like SMA (Simple Moving Average) or EMA
+ * (Exponential Moving Average), while avoiding much of the lag associated with
+ * those averages.
+ *
+ */
+public class JMAIndicator extends CachedIndicator<Num> {
+
+    private final Indicator<Num> indicator; // Base indicator (e.g., Close Price)
+    private final int barCount; // Period for the JMA calculation
+    private final Num phase; // Phase adjustment (usually between -100 and +100)
+    private final Num power; // Smoothing power factor (default is 2)
+    private final Map<Integer, JmaData> jmaDataMap;
+    private final NumFactory numFactory;
+    private final Num beta;
+    private final Num phaseRatio;
+    private final Num alpha;
+
+    /**
+     * Constructor.
+     *
+     * @param indicator The input indicator (e.g., Close Price).
+     * @param barCount  The number of periods for the JMA.
+     * @param phase     The phase adjustment (-100 to +100).
+     * @param power     The smoothing power factor (default is 2).
+     */
+    public JMAIndicator(Indicator<Num> indicator, int barCount, double phase, double power) {
+        super(indicator.getBarSeries());
+        this.numFactory = indicator.getBarSeries().numFactory();
+        this.indicator = indicator;
+        this.barCount = barCount;
+        this.phase = numFactory.numOf(Math.min(Math.max(phase, -100), 100)); // Clamp phase between -100 and 100
+        this.power = numFactory.numOf(Math.max(1, power)); // Ensure power is at least 1
+        this.jmaDataMap = new HashMap<>();
+
+        // Compute smoothing factor based on phase
+        beta = numFactory.numOf(0.45)
+                .multipliedBy(numFactory.numOf(barCount - 1))
+                .dividedBy(numFactory.numOf(0.45).multipliedBy(numFactory.numOf(barCount - 1)).plus(numFactory.two()));
+
+        phaseRatio = this.phase.isLessThan(numFactory.numOf(-100)) ? numFactory.numOf(0.5)
+                : (this.phase.isGreaterThan(numFactory.numOf(100)) ? numFactory.numOf(2.5)
+                        : this.phase.dividedBy(numFactory.numOf(100)).plus(numFactory.numOf(1.5)));
+
+        alpha = beta.pow(this.power);
+
+        for (int i = 0; i < indicator.getBarSeries().getBarCount(); i++) {
```

Comment:
do all barSeries start from zero? I usually do:
``for (int i = barSeries.getStartIndex(); i <= barSeries.getEndIndex(); i++) {``
the existance of getStartIndex() made me think that they could start from something different than 0, but I am not sure.

Reply:
This and the above change should be the last two. I know it's not always possible but reviewers should aim to capture _all_ items within a single review.  There should not be multiple round trips unless subsequent comments are on changes just implemented.

Reply:
I made these changes but I had to go with @francescopeloi approach since I don't have index at that point, anyway I had to use `BarSeries.getBeginIndex()`.   There is no `getStartIndex()`.